openedx_events/event_bus/tests/test_loader.py

```diff
@@ -0,0 +1,118 @@
+        assert loaded == {'def': 'ault'}
+
+    @override_settings(EB_LOAD_PATH='builtins.dict')
+    def test_success(self):
+        with assert_warnings([]):
+            loaded = _try_load(
+                setting_name="EB_LOAD_PATH", args=(), kwargs={'2': 3},
+                expected_class=dict, default={'def': 'ault'},
+            )
+        assert loaded == {'2': 3}
+
+    @override_settings(EB_LOAD_PATH='builtins.list')
+    def test_success(self):
+        with assert_warnings(["builtins.list from EB_LOAD_PATH returned unexpected type <class 'list'>"]):
+            loaded = _try_load(
+                setting_name="EB_LOAD_PATH", args=([1, 2, 3],), kwargs={},
+                expected_class=dict, default={'def': 'ault'},
+            )
+        assert loaded == {'def': 'ault'}
+
+    @override_settings(EB_LOAD_PATH='no_module_here.foo.nope')
+    def test_missing_module(self):
+        with assert_warnings([
+                "Failed to load <class 'dict'> from setting EB_LOAD_PATH: "
+                "ModuleNotFoundError(\"No module named 'no_module_here'\")"
+        ]):
+            loaded = _try_load(
+                setting_name="EB_LOAD_PATH", args=(1), kwargs={'2': 3},
+                expected_class=dict, default={'def': 'ault'},
+            )
+        assert loaded == {'def': 'ault'}
+
+    @override_settings(EB_LOAD_PATH='builtins.does_not_exist')
+    def test_missing_attribute(self):
+        with assert_warnings([
+                "Failed to load <class 'dict'> from setting EB_LOAD_PATH: "
+                "ImportError('Module \"builtins\" does not define a \"does_not_exist\" attribute/class')"
+        ]):
+            loaded = _try_load(
+                setting_name="EB_LOAD_PATH", args=(1), kwargs={'2': 3},
+                expected_class=dict, default={'def': 'ault'},
+            )
+        assert loaded == {'def': 'ault'}
+
+    @override_settings(EB_LOAD_PATH='builtins.dict')
+    def test_bad_args_for_callable(self):
+        with assert_warnings([
+                "Failed to load <class 'dict'> from setting EB_LOAD_PATH: "
+                "TypeError('type object argument after * must be an iterable, not int')"
+        ]):
+            loaded = _try_load(
+                setting_name="EB_LOAD_PATH", args=(1), kwargs={'2': 3},
+                expected_class=dict, default={'def': 'ault'},
+            )
+        assert loaded == {'def': 'ault'}
+
+
+def TestProducer(TestCase):
+
+    @override_settings(EVENT_BUS_PRODUCER=None)
+    def test_default_does_nothing(self):
+        """
+        Test that the default is of the right class but does nothing.
+        """
+        producer = _make_producer()
+
+        with assert_warnings([]):
+            # Nothing thrown, no warnings.
+            assert producer.send(
+                signal=SESSION_LOGIN_COMPLETED, topic='user-logins',
+                event_key_field='user.id', event_data={},
+            ) is None
+
+
+def TestConsumer(TestCase):
```

Comment:
Wow, the coverage check was important here -- this was supposed to be a class. :grimacing: